src/components/CodeInput/PCodeInput.vue

```diff
@@ -86,24 +107,61 @@
     },
   }))
 
+  const styles = computed(() => {
+    return {
+      textarea: {
+        height: `${lineHeight.value * lines.value}px`,
+        width: `${viewWidth.value}px`,
+      },
+      view: {
+        height: `${lineHeight.value * lines.value}px`,
+        width: `${viewWidth.value}px`,
+      },
+    }
+  })
+
+  const updateViewWidth = (): void => {
+    if (textarea.value && source.value) {
+      viewWidth.value = Math.max(textarea.value.scrollWidth, source.value.scrollWidth)
```

Comment:
I didn't notice the resizing issue (though I can reproduce) but this stuck out to me. What is this width fixing?

Reply:
The overscroll is because it's using the starting width as the max width - largely I think this is fine behavior (though I'm open to suggestions) and is only really present on resizable containers

Reply:
I might take a stab at this. Because this would effect anyone resizing their window. Maybe we can huddle and you can explain the issue/thought process here?

Reply:
So textarea elements are pretty annoying - without JS, it gets pretty hacky to set sizing based on content. Unless we set an explicit height/width on the container, we don't have a way of both ensuring that the text area fills the contents of the base input and allows overscrolling when contents exceed the max width of the base input (without resizing the base input.

Today, if you have an empty PCodeInput, the text and view areas fall back to defined min height and width, which is what causes things like `placeholder` attributes to be cut off. There isn't a way (that I'm aware) to support this nicely with CSS alone while accounting for scrollbars, line wrapping, and overflow

Reply:
This is some placeholder text in the current version:
![Screenshot 2023-04-26 at 10 37 29 AM](https://user-images.githubusercontent.com/27291717/234610793-bf690b69-96ba-4ce5-841a-6a532f546284.png)

This is what it should be:
![Screenshot 2023-04-26 at 10 38 09 AM](https://user-images.githubusercontent.com/27291717/234610879-1a81403e-5efe-4e77-8cf6-0707764540d9.png)